### ISSUE TITLE
failed to connect: LDAP Result Code 200 \"Network Error\": dial tcp :389: connect: connection refused

### ISSUE THREAD

**Issue by MoeBensu** [dexidp/dex#3671]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/dexidp/dex/blob/master/.github/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/dexidp/dex/issues) for an issue that matches the one I want to file, without success.
- [X] I am not looking for support or already pursued the available [support channels](https://github.com/dexidp/dex/issues/new/choose) without success.

### Version

v2.40.0 - v2.41.0 (master)

### Storage Type

Postgres

### Installation Type

Official container image

### Expected Behavior

The LDAP connector should be able to connect to the LDAP in a cluster environment.

### Actual Behavior

I use currently the latest image from master (last time built was last Friday) which includes many updates and changes as a preparation for the next release. I tried the image in a cluster setup and I ran into this issue:

```code
failed to connect: LDAP Result Code 200 \"Network Error\": dial tcp :389: connect: connection refused
```

After a closer look, it seems that the connector can't dial up the LDAP server. 

An example of dex config:

```yaml
...
connectors:
- type: ldap
  name: LDAP
  id: ldap
  config:
      host: ldap.core
      insecureNoSSL: true
      insecureSkipVerify: false
...
```

The LDAP server is pingable from within the dex container.  

I first suspected that it could have to do [with replacing `Dial` with `DialURL`](https://github.com/dexidp/dex/commit/63f4410ac1cff2f478b7deb806a386195de4c1b1#diff-b877a4fe9a6210f40c1a5fbcf4c6dd366856b31ec0661975dcf60915cf3985d1). However, the documentation of the new method doesn't give me a clear hint of anything causing this behavior. I couldn't find something that could be the reason behind it in [ the release notes of other updated dependcies like gomplate](https://github.com/hairyhenderson/gomplate/releases/tag/v4.0.0). 

When switching to older dex versions, i.e. v2.38.0, the dex was then able to communicate with the LDAP!! 

In short, it seems that the dex is not able to communicate with LDAP in a cluster setup in the first place. Are there new changes in the dex config that I need to consider in a cluster setup?

### Steps To Reproduce

I also tried to reproduce it locally but I only got the same error when I turned off LDAP. Sending any request to dex even with wrong credentials would throw this:

<img width="1103" alt="Screenshot 2024-08-04 at 23 10 51" src="https://github.com/user-attachments/assets/7a68d1b8-2e98-44c4-bb7e-52eeff1a797e">

### Logs

```shell
Turning on the debug  didn't give more information.
```

**Comment by nemunaire** [comment]
I have the same problem with the Docker container since the update to v2.41.0.

In my case, the error message is:
> Login error: failed to connect: LDAP Result Code 200 "Network Error": dial tcp :636: connect: connection refused
(I don't have the option `insecureNoSSL: true`)

Regardless of the `host` setting, the error remains the same. I tried several host/port combinations, but the error message persists.

I rolled back to container tag `v2.40.0` and it works well.

**Comment by justusbunsi** [comment]
Same issue for me. LDAPS with port 636, no `insecure...` option enabled. Rollback to 2.40.0 worked for mee, too.

**Comment by nabokihms** [comment]
It seems our integrations tests do not cover different ssl option combinations… I will try to investigate, thank you all for reporting.

**Comment by nabokihms** [comment]
Ok, I found the problem, and it is with the schema. Funny one. It will be fixed in v2.41.1
https://github.com/go-ldap/ldap/issues/523#issuecomment-2269844629
